R/calcHDDCDD.R

```diff
@@ -1,101 +1,714 @@
-                 })
-
-  yearsScenario <- c(gsub("y", "", attributes(index)$dimnames$period[index]) %>% as.integer(), 2100)
-  yearsHistory <- setdiff(getYears(data, as.integer = TRUE), yearsScenario)
-
-  convergenceYear <- 2030
-  convergenceYears <- c(max(yearsHistory), yearsScenario[yearsScenario <= convergenceYear])
-  afterConvergenceYears <- yearsScenario[!yearsScenario %in% convergenceYears]
-  lambda <- c(rep(1, length(yearsHistory)),
-              tail(seq(1, 0, length.out = length(convergenceYears)), -1),
-              rep(0, length(afterConvergenceYears)))
-  names(lambda) <- c(yearsHistory, yearsScenario)
-  lambda <- as.magpie(lambda)
-
-  rcps     <- grep("history", getNames(data, TRUE)$rcp,      value = TRUE, invert = TRUE)
-  ssps     <- grep("history", getNames(data, TRUE)$scenario, value = TRUE, invert = TRUE)
-  variable <- grep("history", getNames(data, TRUE)$variable, value = TRUE, invert = TRUE)
-
-  data[is.na(data)] <- 0
-  data <- mbind(data, (1 / 3) * (setYears(data[, 2099, ], 2100) +
-                                 setYears(data[, 2098, ], 2100) +
-                                 setYears(data[, 2097, ], 2100))) # give a point for 2100
-
-  # distribute the history between all scenarios
-  data <- do.call("mbind", lapply(rcps, function(rcp) {
-    do.call("mbind", lapply(ssps, function(ssp) {
-      tmp <- data[, , rcp][, , ssp]
-      tmp[, yearsHistory, ] <- setNames(data[, yearsHistory, "history"][, , "history"],
-                                        getNames(tmp[, yearsHistory, ]))
+#' @author Robin Krekeler, Hagen Tockhorn
+#' @examples
+#'
+#' \dontrun{
+#' calcHDDCDD()
+#' }
+#'
+#' @importFrom madrat getConfig
+#' @importFrom raster cellStats
+#' @importFrom ncdf4 nc_open
+#' @importFrom tidyr %>%
+#' @importFrom dplyr mutate
+#' @importFrom rlang .data
+#' @importFrom pracma integral2
+#' @importFrom stringr str_sub
+#' @importFrom terra app nlyr tapp subset rast classify time
+
+
+calcHDDCDD <- function(mappingFile, bait=FALSE, multiscen = FALSE) {
+
+
+  # FUNCTIONS-------------------------------------------------------------------
+
+  # fill dates for unnamed data
+  #   ---> was supposed to go to convertISIMIP, doesn't work with country check
+  fillDates <- function(r, filename, bait=FALSE) {
+    if (grepl(".nc|.nc4", filename)) {
+      filename <- gsub(".nc|.nc4", "", filename)
+    }
+
+    if (bait) {
+      dStart <- as.Date(stringr::str_sub(filename, -17, -10),
+                        format = "%Y%m%d")
+      n <- nlyr(r)
+      dates <- seq.Date(dStart, by = "day", length.out = n)
+    }
+    else {
+      yStart <- stringr::str_sub(filename, -9, -6)
+      dStart <- as.Date(paste0(yStart, "-1-1"))
+      n <- nlyr(r)
+
+      dates <- seq.Date(dStart, by = "day", length.out = n)
+    }
+
+    # fill dates
+    names(r) <- dates
+    return(r)
+  }
+
+
+  # prepare input for calcBAIT()
+  calcBaitInput <- function(frsds=NULL, fsfc=NULL, fhuss=NULL, baitInput=NULL, mean=FALSE) {
```

Comment:
I could need some more comments in this function to understand what happens. Also try to use indicative argument names like `fillWithMean` or `fillGaps` instead of `mean`. This can be as effective as three lines of in-code documentation.